packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2709,71 +2707,70 @@ class RenderMouseRegion extends RenderProxyBox {
+  @override
   PointerEnterEventListener get onEnter => _onEnter;
+  PointerEnterEventListener _onEnter;
   set onEnter(PointerEnterEventListener value) {
     if (_onEnter != value) {
       _onEnter = value;
       _markPropertyUpdated(mustRepaint: false);
     }
   }
-  PointerEnterEventListener _onEnter;
-  void _handleEnter(PointerEnterEvent event) {
-    if (_onEnter != null)
-      _onEnter(event);
-  }
 
   /// Called when a pointer changes position without buttons pressed and the end
   /// position is within the region.
+  @override
   PointerHoverEventListener get onHover => _onHover;
+  PointerHoverEventListener _onHover;
   set onHover(PointerHoverEventListener value) {
     if (_onHover != value) {
       _onHover = value;
       _markPropertyUpdated(mustRepaint: false);
     }
   }
-  PointerHoverEventListener _onHover;
-  void _handleHover(PointerHoverEvent event) {
-    if (_onHover != null)
-      _onHover(event);
-  }
 
   /// Called when a pointer is no longer contained by the region (with or
   /// without buttons pressed) for any reason.
-  ///
-  /// This callback is always matched by an earlier [onEnter].
-  ///
-  /// See also:
-  ///
-  ///  * [MouseRegion.onExit], which uses this callback, but is not triggered in
-  ///    certain cases and does not always match its earier [MouseRegion.onEnter].
+  @override
   PointerExitEventListener get onExit => _onExit;
+  PointerExitEventListener _onExit;
   set onExit(PointerExitEventListener value) {
     if (_onExit != value) {
       _onExit = value;
       _markPropertyUpdated(mustRepaint: false);
     }
   }
-  PointerExitEventListener _onExit;
-  void _handleExit(PointerExitEvent event) {
-    if (_onExit != null)
-      _onExit(event);
-  }
 
-  // Object used for annotation of the layer used for hover hit detection.
-  MouseTrackerAnnotation _hoverAnnotation;
+  /// The mouse cursor for mouse pointers that are hovering over the annotated
+  /// region.
+  ///
+  /// When a mouse enters the region, its cursor will be changed to the
+  /// [cursor]. The [cursor] defaults to null, meaning the region does not
+  /// control cursors, but defers the choice to the next region behind this
+  /// one on the screen in hit-test order, or [SystemMouseCursors.basic] if no
+  /// others can be found.
+  ///
+  /// See also:
+  ///
+  ///  * [MouseCursor] for a general introduction to the mouse cursor system.
+  ///  * [MouseRegion.cursor], which is the widget-layer counterpart, and
+  ///    allows general (unprepared) mouse cursors.
+  @override
+  MouseCursor get cursor => _cursor;
+  MouseCursor _cursor;
+  set cursor(MouseCursor value) {
+    if (_cursor != value) {
+      _cursor = value;
+      _markPropertyUpdated(mustRepaint: true);
+    }
+  }
 
   /// Object used for annotation of the layer used for hover hit detection.
   ///
   /// This is only public to allow for testing of Listener widgets. Do not call
   /// in other contexts.
   @visibleForTesting
```

Comment:
Yeah, but that will be a breaking change. I'll do it in another PR, along with some other deprecated APIs.

Reply:
Is this used in any of the customer_tests or internally? Otherwise its not considered breaking. I would just clean this up right now to avoid more tech debt.